### ISSUE TITLE
Rename "DelegateEndpoint" types

### ISSUE THREAD

**Issue by halter73** [dotnet/aspnetcore#36553]
Given that the options type related to the `Map(string, Delegate)`-style methods is named `RouteHandlerOptions`, we should make the following more consistent.

- `DelegateEndpointConventionBuilder` -> `RouteHandlerBuilder`
- `OpenApiDelegateEndpointConventionBuilderExtensions` -> `OpenApiRouteHandlerBuilderExtensions`
- `DelegateEndpointRouteBuilderExtensions` merges with the existing `EndpointRouteBuilderExtensions`.

Basically, this changes `DelegateEndpoint` to `RouteHandler` and removes `Convention` in the class names.
